diff --git a/dropbearsign.c b/dropbearsign.c
index 8c3dbcb..16405a7 100644
--- a/dropbearsign.c
+++ b/dropbearsign.c
@@ -41,13 +41,13 @@ static void sign_one(sign_key *key, const char *filename, int fd,
 	ssize_t data_len;
 	hash_state hs;
 	const struct ltc_hash_descriptor *hash_desc = &sha512_desc;
-	FILE *outfile;
+	FILE *outfile = NULL;
 	char *outname;
 	buffer *buf = NULL;
 	unsigned int buf_len;
 	buffer *sig = NULL;
 	unsigned int sig_len;
-	int ret;
+	int ret = 1;
 
 	buf_len = sizeof MAGIC_PREAMBLE - 1 +
 		4 + strlen(namespace) +
@@ -103,6 +103,7 @@ static void sign_one(sign_key *key, const char *filename, int fd,
 		if (!outfile) {
 			fprintf(stderr, "Error opening signature file '%s'\n",
 				outfilename);
+			goto out;
 		}
 		m_free(outfilename);
 	}
