From 64991e974ec63e4e07ab7c52627a890d7186ad62 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Thu, 21 Apr 2022 13:53:40 +0200
Subject: [PATCH] fix fsck for toybox not supporting -a flag

---
 lib/init/rc.boot | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/init/rc.boot b/lib/init/rc.boot
index bb38e92..2fba3bf 100755
--- a/lib/init/rc.boot
+++ b/lib/init/rc.boot
@@ -54,7 +54,7 @@ log "Remounting rootfs as read-only..."; {
 }
 
 log "Checking filesystems..."; {
-    fsck -ATat noopts=_netdev
+    fsck -ATt noopts=_netdev
 
     # It can't be assumed that success is 0
     # and failure is > 0.
-- 
2.36.0

