--- config.def.h
+++ config.def.h
@@ -8,9 +8,9 @@
 
 struct rule rules[] = {
 	/* sequence     event        lines */
-	{"\033[5;2~",   SCROLL_UP,   -1},	/* [Shift] + [PageUP] */
-	{"\033[6;2~",   SCROLL_DOWN, -1},	/* [Shift] + [PageDown] */
+	{"\033[5;2~",   SCROLL_UP,   5},	/* [Shift] + [PageUP] */
+	{"\033[6;2~",   SCROLL_DOWN, 5},	/* [Shift] + [PageDown] */
 	/* mouse binding shadows ^E and ^Y, so it's disabled by default */
-	//{"\031",        SCROLL_UP,    1},	/* mouse wheel up */
-	//{"\005",        SCROLL_DOWN,  1},	/* mouse wheel Down */
+	{"\031",        SCROLL_UP,    1},	/* mouse wheel up */
+	{"\005",        SCROLL_DOWN,  1},	/* mouse wheel Down */
 };
