From 0149ac4bd9ac49ff2f42a6b79056eaf0375ba244 Mon Sep 17 00:00:00 2001
From: hovercats <hovercatswithlasereyes@protonmail.com>
Date: Mon, 28 Aug 2023 21:51:35 +0000
Subject: [PATCH 5/5] switch colorscheme

---
 config.def.h | 51 ++++++++++++++++++++++-----------------------------
 1 file changed, 22 insertions(+), 29 deletions(-)

diff --git a/config.def.h b/config.def.h
index bd75078..baa0429 100644
--- a/config.def.h
+++ b/config.def.h
@@ -101,42 +101,35 @@ unsigned int tabspaces = 8;
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
 	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
+ [0] = "#222222", /* black    */
+ [1] = "#8b3e2f", /* red      */
+ [2] = "#526f33", /* green    */
+ [3] = "#665847", /* yellow   */
+ [4] = "#4a708b", /* blue     */
+ [5] = "#7a378b", /* magenta  */
+ [6] = "#528b8b", /* cyan     */
+ [7] = "#999999", /* white    */
 
 	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
-
-	[255] = 0,
-
-	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
-	"gray90", /* default foreground colour */
-	"black", /* default background colour */
-};
+ [8]  = "#4c4c4c", /* black   */
+ [9]  = "#d75f00", /* red     */
+ [10] = "#cee318", /* green   */
+ [11] = "#eee685", /* yellow  */
+ [12] = "#9ac0cd", /* blue    */
+ [13] = "#9f79ee", /* magenta */
+ [14] = "#79cdcd", /* cyan    */
+ [15] = "#e8e9ca", /* white   */
+
 
+};
 
 /*
  * Default colors (colorname index)
- * foreground, background, cursor, reverse cursor
+ * foreground, background, cursor
  */
-unsigned int defaultfg = 258;
-unsigned int defaultbg = 259;
-unsigned int defaultcs = 256;
+unsigned int defaultfg = 15;
+unsigned int defaultbg = 0;
+unsigned int defaultcs = 15;
 static unsigned int defaultrcs = 257;
 
 /*
-- 
2.42.0

